bin/plugin/commands/benchmark.js

```diff
@@ -0,0 +1,231 @@
+	}
+
+	outputResults( opt, results );
+};
+
+/**
+ * Generates URLs to benchmark based on command arguments. If both "<url>" and "<file>" arguments
+ * are passed to the command, then both will be used to generate URLs.
+ *
+ * @param {BenchmarkCommandOptions} opt Command options.
+ */
+async function* getURLs( opt ) {
+	if ( !! opt.url ) {
+		yield opt.url;
+	}
+
+	if ( !! opt.file ) {
+		const rl = readline.createInterface( {
+			input: fs.createReadStream( opt.file ),
+			crlfDelay: Infinity,
+		} );
+
+		for await ( const url of rl ) {
+			if ( url.length > 0 ) {
+				yield url;
+			}
+		}
+	}
+}
+
+/**
+ * Benchmarks an URL and returns response time and server-timing metrics for every request.
+ *
+ * @param {BenchmarkOptions} params Benchmark parameters.
+ * @return {BenchmarkResults} Response times and metrics arrays.
+ */
+function benchmarkURL( params ) {
+	const metrics = {};
+	const responseTimes = [];
+
+	const onHeaders = ( { headers } ) => {
+		const responseMetrics = getServerTimingMetricsFromHeaders( headers );
+		Object.entries( responseMetrics ).forEach( ( [ key, value ] ) => {
+			metrics[ key ] = metrics[ key ] || [];
+			metrics[ key ].push( +value );
+		} );
+	};
+
+	const onResponse = ( statusCode, resBytes, responseTime ) => {
+		responseTimes.push( responseTime );
+	};
+
+	const instance = autocannon( {
+		...params,
+		setupClient( client ) {
+			client.on( 'headers', onHeaders );
+			client.on( 'response', onResponse );
+		},
+	} );
+
+	const onStop = instance.stop.bind( instance );
+	process.once( 'SIGINT', onStop );
+
+	return new Promise( ( resolve ) => {
+		instance.on( 'done', () => {
+			process.off( 'SIGINT', onStop );
+			resolve( { responseTimes, metrics } );
+		} );
+	} );
+}
+
+/**
+ * Reads the Server-Timing metrics from the response headers.
+ *
+ * @param {Array.<string>} headers Array of response headers information where each even element is a header name and an odd element is the header value.
+ * @return {Object} An object where keys are metric names and values are metric values.
+ */
+function getServerTimingMetricsFromHeaders( headers ) {
+	for ( let i = 0, len = headers.length; i < len; i += 2 ) {
+		if ( headers[ i ].toLowerCase() !== 'server-timing' ) {
+			continue;
+		}
+
+		return headers[ i + 1 ]
+			.split( ',' )
+			.map( ( timing ) => timing.trim().split( ';dur=' ) )
+			.reduce(
+				( obj, [ key, value ] ) => ( { ...obj, [ key ]: value } ),
+				{}
+			);
+	}
+
+	return {};
+}
+
+/**
+ * Ouptuts results of benchmarking.
+ *
+ * @param {BenchmarkCommandOptions} opt     Command options.
+ * @param {Array.<Array>}           results A collection of benchmark results for each URL.
+ */
+function outputResults( opt, results ) {
+	const len = results.length;
+	const allMetricNames = {};
+
+	const newRow = ( title ) => {
+		const line = new Array( len + 1 ).fill( '' );
+		line[ 0 ] = title;
+		return line;
+	};
+
+	const tableData = [ newRow( '' ), newRow( 'Avg Time' ) ];
```

Comment:
Maybe use `Response Time` instead? The average is used for every metric, so that's a bit pointless to mention here (plus see my comment below that we should be using median though).

Reply:
Updated.